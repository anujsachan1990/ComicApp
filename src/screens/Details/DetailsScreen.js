import React, { useState, useEffect, memo } from 'react'
import { StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import { Content, View } from 'native-base'
import MainContainer from '../../components/MainContainer/MainContainer'
import MainActivityIndicator from '../../components/MainActivityIndicator/MainActivityIndicator'
import Card from '../../components/Card/Card'
import apiService from '../../services/apiURL'
import theme from '../../styles/theme'
import utils from '../../utilities'

const styles = StyleSheet.create({
  content: {
    backgroundColor: theme.color.primaryColor,
    padding: 20,
    margin: 20,
    borderRadius: 10,
    minHeight: 300,
  },
})

const DetailsScreen = props => {
  const [isLoading, setLoading] = useState(true)
  const [comicDetail, setDetail] = useState({})

  useEffect(() => {
    const { navigation } = props
    const itemId = navigation.state.params.item.diamond_id

    utils.fetchData(`${apiService.getDetailByDiamondId}${itemId}`).then(data => {
      setDetail({ comicDetail: data })
      setLoading(false)
    })
  }, [props])

  const getcontent = () => {
    return isLoading ? (
      <MainActivityIndicator />
    ) : (
      <View>
        {comicDetail.comicDetail &&
          comicDetail.comicDetail.comics.map(item => {
            return <Card key={item.diamond_id} item={item} />
          })}
      </View>
    )
  }

  return (
    <MainContainer>
      <Content padder contentContainerStyle={isLoading ? styles.content : {}}>
        {getcontent()}
      </Content>
    </MainContainer>
  )
}

DetailsScreen.propTypes = {
  navigation: PropTypes.shape({
    state: PropTypes.shape({
      params: PropTypes.shape({
        item: PropTypes.shape({
          diamond_id: PropTypes.string.isRequired,
        }),
      }),
    }),
  }).isRequired,
}

export default memo(DetailsScreen)
