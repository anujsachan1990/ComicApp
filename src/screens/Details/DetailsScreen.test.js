import React from 'react'
import renderer from 'react-test-renderer'

import DetailsScreen from './DetailsScreen'

const props = {
  navigation: {
    state: {
      param: {
        item: {
          diamond_id: 'someID',
        },
      },
    },
  },
}

describe('<DetailsScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DetailsScreen {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
