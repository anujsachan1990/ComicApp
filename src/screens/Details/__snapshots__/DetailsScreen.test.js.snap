// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsScreen /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#F3F3F3",
      },
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": 10,
        },
        Object {
          "backgroundColor": "#00838d",
          "borderRadius": 10,
          "margin": 20,
          "minHeight": 300,
          "padding": 20,
        },
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    padder={true}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
